tests/functionnal/CommonDBTM.php

```diff
@@ -114,4 +114,62 @@ public function testgetIndexNameOtherThanID() {
       ]))->isFalse();
 
    }
+
+   /**
+    * @engine inline
+    */
+   public function testGetFromDBByCrit() {
+      $instance = new \Computer();
+      $instance->getFromDBByCrit(['name' => '_test_pc01']);
+      // the instance must be populated
+      $this->boolean($instance->isNewItem())->isFalse();
+
+      $instance = new \Computer();
+      $this->exception(
+         function() use ($instance) {
+            $instance->getFromDBByCrit(['entities_id' => '1']);
+         }
+      );
```

Comment:
Maybe test exception class or message?

Reply:
An exception is thrown only in the unit tests context, t hen UI considered useless to test its class or message.

I would more easily agree if getFromDBByCrit() or getFromDBByRequest() explicitly throw an exception. I did not t ook  the freedom to do it, and preferred to keep the original behaviour.

Reply:
The exception produces a message similar to this

```CommonDBTM::getFromDBByRequest() in /home/intra.dethegeek.eu.org/dethegeek/public_html/glpi-flyvemdm-93/inc/commondbtm.class.php line 311
getFromDBByRequest expects to get one result, 2 found! ```

The line of code is not predictible and may change often. I'll test  the type of exception only, as I don"'t see an obvious way to get the message and test a substring only.

Reply:
See https://github.com/glpi-project/glpi/blob/9.2/bugfixes/tests/units/DBmysqlIterator.php#L52